src/main/java/io/stargate/sgv2/jsonapi/service/schema/SourceModel.java

```diff
@@ -0,0 +1,107 @@
+package io.stargate.sgv2.jsonapi.service.schema;
+
+import io.stargate.sgv2.jsonapi.exception.ErrorCodeV1;
+import io.stargate.sgv2.jsonapi.exception.JsonApiException;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * The source model used for the vector index. This is only applicable if the vector index is
+ * enabled.
+ */
+public enum SourceModel {
+  ADA002("ada002"),
+  BERT("bert"),
+  COHERE_V3("cohere-v3"),
+  GECKO("gecko"),
+  NV_QA_4("nv-qa-4"),
+  OPENAI_V3_LARGE("openai-v3-large"),
+  OPENAI_V3_SMALL("openai-v3-small"),
+  OTHER("other"),
+  UNDEFINED("undefined");
+
+  private final String name;
+  public static final Map<String, SourceModel> SOURCE_MODEL_NAME_MAP =
```

Comment:
Would be better to be `private` and only access via "get" or "find" method for lookups.
Unless the whole `Map` is needed for some reason?